File: wrappers/javascript/anoncreds-shared/src/api/KeyCorrectnessProof.ts
import type { JsonObject } from '../types'

import { AnoncredsObject } from '../AnoncredsObject'
import { anoncreds } from '../register'

export class KeyCorrectnessProof extends AnoncredsObject {
  public static fromJson(json: JsonObject) {
    return new KeyCorrectnessProof(anoncreds.keyCorrectnessProofFromJson({ json: JSON.stringify(json) }).handle)
  }
}

File: wrappers/dotnet/lib/Models/KeyCorrectnessProof.cs
namespace AnonCredsNet.Models;

public class KeyCorrectnessProof : AnonCredsObject
{
    internal KeyCorrectnessProof(long handle)
        : base(handle) { }

    internal static KeyCorrectnessProof FromJson(string json) =>
        FromJson<KeyCorrectnessProof>(json);
}
